### ISSUE TITLE
[LOW] [Splits] [$500] [MEDIUM] Web - IOU - Receipt from manual request remains on Admins account when Employee delete it

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#35208]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.32-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Employee creates manual request on collect workspace chat
2. Employee add a receipt to IOU
3. Employee delete the receipt
4. Admin open the IOU request

## Expected Result:
Receipt should be not presented on Admin's side

## Actual Result:
After deletion receipt is still present until refresh the page

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/6e57b4f4-cd0a-419d-b26e-78ab1dbabed8

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~014fe31ab963ba8b49</li>
        <li>Upwork Job ID: 1750654064488505344</li>
        <li>Last Price Increase: 2024-02-15</li>
    </ul>
</details>

**Comment by lanitochka17** [comment]
We think that this bug might be related to #wave5. 
CC @dylanexpensify

**Comment by MitchExpensify** [comment]
Wave 5 worthy @dylanexpensify ?

**Comment by MitchExpensify** [comment]
Friendly bump @dylanexpensify

**Comment by MitchExpensify** [comment]
Checking in with @dylanexpensify 👀

**Comment by dylanexpensify** [comment]
DMing back!

**Comment by rezkiy37** [comment]
Hi, I’m Michael (Mykhailo) from Callstack and I would like to work on this issue.

**Comment by rezkiy37** [comment]
I've investigated the bug. I can confirm that it is still replicable. I see that we actively use the **Pusher** for this logic. So, the **Pusher** does not send an update for an admin, when a member removed a receipt. Also, even after a re-fetching data, it is does not overwrite a `receipt` property of a `transaction`. It resets a `filename` property, but it is not enough. 
So, after this, I can say that someone should handle this from the backend side.

https://github.com/Expensify/App/assets/57314631/3896e7a3-38c5-4155-9aaa-a6b290afd813

**Comment by rezkiy37** [comment]
Still waiting on some feedback about a comment above.

**Comment by MitchExpensify** [comment]
Who are you waiting for feedback from @rezkiy37 ?

**Comment by rezkiy37** [comment]
We need someone from the backend team to clarify [here](https://github.com/Expensify/App/issues/35208#issuecomment-1934535903).

**Comment by MitchExpensify** [comment]
Is there a specific backend member we should ping for [this](https://github.com/Expensify/App/issues/35208#issuecomment-1943460686) @rezkiy37 ?

**Comment by rezkiy37** [comment]
> Is there a specific backend member we should ping for [this](https://github.com/Expensify/App/issues/35208#issuecomment-1943460686) @rezkiy37 ?

Not a specific one. We need anybody who can help us here from the backend side.

**Comment by MitchExpensify** [comment]
@dylanexpensify Anyone close to Wave 6 that might be our backend hero here? https://github.com/Expensify/App/issues/35208#issuecomment-1934535903

**Comment by MitchExpensify** [comment]
Perhaps the wave 5 team engineer? @dylanexpensify

**Comment by MitchExpensify** [comment]
Chatted with @dylanexpensify and we're thinking this may in fact be a wave 8 related issue, what do you think @zanyrenney ?

**Comment by MitchExpensify** [comment]
Double checking this is still reproducible..

*Chrome**

I think everything is working as should be expected except I'm not seeing the real time updates like OP seems to be. I have to hard refresh to see the receipt appear on the admin end. 

Once I do refresh and see the receipt has been deleted, if I click the space where the receipt was.. I get the endless spinner we should fix here:

<img width="1052" alt="image" src="https://github.com/Expensify/App/assets/36425901/628dbaae-92fd-431e-a064-7a2dd158d664">





**Safari**

Receipt doesn't show for admin when I add a receipt at all until I hard refresh:

<img width="1711" alt="image" src="https://github.com/Expensify/App/assets/36425901/e90c2803-e60c-48a6-a7b5-6227bfb9f0c1">

<img width="1694" alt="image" src="https://github.com/Expensify/App/assets/36425901/30a67217-14b9-4f84-aab2-627cb3b7f532">

When I delete and hard refresh it looks like this:

<img width="1711" alt="image" src="https://github.com/Expensify/App/assets/36425901/d2997cf0-3243-455b-aa11-04b0d8edf9ff">

**Comment by MitchExpensify** [comment]
Two bugs:

1. Receipt not deleting until hard refresh 
2. Endless spinner when clicking the receipt section after hard refresh

@mountiny tapping you as a buddy check on 1.. _Should_ we be able to see that the receipt is deleted without a refresh?

**Comment by mountiny** [comment]
I think we had a similar issues with images in chat being deleted. I think we should be able to see this dynamically, it might have to be internal issue, this needs to be checked. We either do not push the updates correctly (internal) or we do not re-render the UI based on the updated data correctly (and you need to refresh)

**Comment by MitchExpensify** [comment]
We probably won't see any progress here while @rezkiy37 is on vacay.

**Comment by rezkiy37** [comment]
As Vit said before https://github.com/Expensify/App/issues/35208#issuecomment-1988420318. Looks like that's an internal issue for the internal devs.

**Comment by MitchExpensify** [comment]
> We either do not push the updates correctly (internal) or we do not re-render the UI based on the updated data correctly (and you need to refresh)

@mountiny do you have any bandwidth to check this out?

**Comment by mountiny** [comment]
Not at the moment, still focusing on Wave collect release 1

**Comment by rezkiy37** [comment]
Should we mark this issue as "On Hold"?

**Comment by mountiny** [comment]
Why on hold? Its ready to be investigated, I have added the Hot Pick label

**Comment by rezkiy37** [comment]
Ah, good. I misunderstood this [comment](https://github.com/Expensify/App/issues/35208#issuecomment-2022863993).

**Comment by MitchExpensify** [comment]
Waiting on Internal volunteer, Melvin

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by MitchExpensify** [comment]
I was unable to reproduce this onev1.4.62-8

**Comment by MitchExpensify** [comment]
@waterim Are you able to reproduce this> I am not currently able to

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Third week)
